### PR TITLE
[18.0][MIG] base_report_to_printer: Migration to 18.0

### PR DESCRIPTION
**Pull request by trisdoan**
### Note
- Missing commits from 14.0 are ported in https://github.com/trisdoan/report-print-send/pull/1
- Tests are failed due to lack of CUPS server configured

### Changes in 18.0
- `ir.property` was removed in https://github.com/odoo/odoo/commit/de302c2d36305c0d7562572a30587641eabfe914. Hence, replace default value for `property_printing_action_id` with invoked function.
- `doall` & `numbercall` fields were removed from ir_cron in https://github.com/odoo/odoo/pull/151628/commits/3a9949adbc87a1ca835aa7688e1f426b0ebbc8a8
- callable is not validated in selection any longer due to performance cost in https://github.com/odoo/odoo/pull/154177/commits/eaef3d28236fed7c72da287f060ae37aa8617f01. So, initialise `AVAILABLE_ACTION_TYPES` with the precomputed value instead of the function which returns the precomputed value.

### This changes
- Take this chance to replace percent format with f-string

### CODE REVIEW

File: base_report_to_printer/models/printing_report_xml_action.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        comodel_name="ir.actions.report",
+        string="Report",
+        required=True,
+        ondelete="cascade",
+    )
+    user_id = fields.Many2one(
+        comodel_name="res.users", string="User", required=True, ondelete="cascade"
+    )
+    action = fields.Selection(
+        selection=lambda s: s.env["printing.action"]._available_action_types(),
```
Comment (nguyenminhchien) [inline base_report_to_printer/models/printing_report_xml_action.py]: Error when visiting `Settings > Printing > Report`
image

File: base_report_to_printer/wizards/printing_printer_update_wizard_view.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    <record id="action_printer_update_wizard" model="ir.actions.act_window">
+        <field name="name">Update Printers from CUPS</field>
+        <field name="res_model">printing.printer.update.wizard</field>
+        <field name="view_mode">form</field>
+        <field name="target">new</field>
+    </record>
+    <menuitem
+        action="action_printer_update_wizard"
+        sequence="40"
+        id="menu_printer_update_wizard"
```
Comment (nguyenminhchien) [inline base_report_to_printer/wizards/printing_printer_update_wizard_view.xml:34]: this menu (Update Printers from CUPS) is useless, no group has `create` right on this wizard, even Admin user will face this error:
image

File: base_report_to_printer/models/printing_action.py [hunk 3]
```diff
@@ -5,22 +5,20 @@
...
-        return [
-            ("server", "Send to Printer"),
-            ("client", "Send to Client"),
-            ("user_default", "Use user's defaults"),
-        ]
-
     name = fields.Char(required=True)
     action_type = fields.Selection(
-        selection=_available_action_types, string="Type", required=True
+        selection=AVAILABLE_ACTION_TYPES, string="Type", required=True
```
Comment (nguyenminhchien) [inline base_report_to_printer/models/printing_action.py:23]: @trisdoan do you have good reason for this change?

File: base_report_to_printer/views/ir_actions_report.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        <field name="name">ir.actions.report.form (in base_report_to_printer)</field>
+        <field name="model">ir.actions.report</field>
+        <field name="inherit_id" ref="base.act_report_xml_view" />
+        <field name="arch" type="xml">
+            <page name="security" position="before">
+                <page string="Print" name="print">
+                    <group>
+                        <field name="property_printing_action_id" />
+                        <field name="printing_printer_id" />
+                        <field name="printer_tray_id" />
```
Comment (nguyenminhchien) [inline base_report_to_printer/views/ir_actions_report.xml:13]: Error when accessing to menu Settings > Technica > .. > Reports

image

**Review by nguyenminhchien — CHANGES_REQUESTED** [review]

File: base_report_to_printer/tests/test_report.py [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+            .create(
+                {
+                    "name": "test_pdf",
+                    "module": "base_report_to_printer",
+                    "model": "ir.ui.view",
+                    "res_id": self.report_pdf_view.id,
+                }
+            )
+        )
+        self.report_text = self.Model.create(
```
Comment (nguyenminhchien) [inline base_report_to_printer/tests/test_report.py]: self.report_text = self.Model.create() is duplicated after porting
Seems that you don't need to port https://github.com/trisdoan/report-print-send/pull/1/commits/fc3ad723f5da1795ad5025d1c32406a71f44cbfa but add it to blacklist 

anw, pick the commit https://github.com/trisdoan/report-print-send/pull/1/commits/a7f1f367f9d4b50992f5d04a7295173ed4661dc8 into this PR as well.

File: base_report_to_printer/wizards/printing_printer_update_wizard_view.xml (hunk 2 above)
Comment (trisdoan) [reply base_report_to_printer/wizards/printing_printer_update_wizard_view.xml:34]: Hi anh, I fixed and included in https://github.com/trisdoan/report-print-send/pull/1

File: base_report_to_printer/tests/test_report.py (hunk 5 above)
Comment (trisdoan) [reply base_report_to_printer/tests/test_report.py]: Hi anh, https://github.com/trisdoan/report-print-send/commit/fc3ad723f5da1795ad5025d1c32406a71f44cbfa was wrongly ported, I fixed and updated the PR

**Review by nguyenminhchien — CHANGES_REQUESTED** [review]
Should squash these commits into a single one: `[MIG] base_report_to_printer: Migration to 18.0`
image

**Review by nguyenminhchien — APPROVED** [review]

**Pull request closed without merging**